### PR TITLE
Added blob url support

### PR DESCRIPTION
**Pull request by andre-f-paggi**
It allows the usage of blob urls as a way to avoid passing '?papaworker' in the querystring and work with difficult webserver routes.

Issue: #426

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,2107 @@
+{
```
Comment (pokoli) [inline package-lock.json]: This file should not be included in the PR

File: papaparse.js [hunk 2]
```diff
@@ -57,6 +57,7 @@
 	Papa.BAD_DELIMITERS = ['\r', '\n', '"', Papa.BYTE_ORDER_MARK];
 	Papa.WORKERS_SUPPORTED = !IS_WORKER && !!global.Worker;
 	Papa.SCRIPT_PATH = null;	// Must be set by your code if you use workers and this lib is loaded asynchronously
+	Papa.SCRIPT_PATH_AS_BLOB_URL = false;	// Should be used when you want to use blob urls as 'SCRIPT_PATH' location.
```
Comment (pokoli) [inline papaparse.js]: I think we should always use one method for loading workers. So for me this variable is useless.

File: papaparse.js [hunk 3]
```diff
@@ -42,7 +42,7 @@
 
 
 	var IS_WORKER = !global.document && !!global.postMessage,
-		IS_PAPA_WORKER = IS_WORKER && /(\?|&)papaworker(=|&|$)/.test(global.location.search),
+		IS_PAPA_WORKER = IS_WORKER && (/(\?|&)papaworker(=|&|$)/.test(global.location.search) || isBlobURL(global.location)),
```
Comment (pokoli) [inline papaparse.js]: Indeed insted of testing if it's blob, we should probably use a diferent url. For example ending with "worker"

File: papaparse.js [hunk 4]
```diff
@@ -1468,15 +1469,46 @@
...
+		}
+
 		var w = new global.Worker(workerUrl);
 		w.onmessage = mainThreadReceivedMessage;
 		w.id = workerIdCounter++;
 		workers[w.id] = w;
 		return w;
 	}
 
+	function createDummyBlobUrlFromUrl(script_path) {
```
Comment (pokoli) [inline papaparse.js]: This function should remove the suffix added to know that's a worker and then load the correct javascript file.

**Comment by pokoli** [comment]
Thanks for the PR, I've added some remarks, could you review them please?

File: papaparse.js (hunk 2 above)
Comment (andre-f-paggi) [reply papaparse.js]: I agree it would be better with just one method, but the blob url support nowadays is not very high.
http://caniuse.com/#feat=bloburls

File: papaparse.js (hunk 3 above)
Comment (andre-f-paggi) [reply papaparse.js]: Humm, you mean the blob url should end with "worker"?
I don't know if it is possible nowadays.

**Comment by andre-f-paggi** [comment]
Thanks for reviewing and helping me out on one of my firsts PRs. :slightly_smiling_face: 
I answered some remarks you made.

File: papaparse.js (hunk 3 above)
Comment (pokoli) [reply papaparse.js]: I mean that we should not decide if it's a worked bassed blob, but using a diferent url. 

I think it's legitime to load paparse using a blob. With the current code it will asume allways it's a worker, which may not be true. That's why I'm sugesting to add a sufix to determine if it's worker or not.

File: papaparse.js (hunk 2 above)
Comment (pokoli) [reply papaparse.js]: I see it's supported on all major browsers. So it's ok for my side.

File: papaparse.js (hunk 3 above)
Comment (andre-f-paggi) [reply papaparse.js]: You are right, but I don't have any other idea since the suffix didn't work for me.

Comment (pokoli) [reply papaparse.js]: Could you explain what you have tested, what is working and what is not working? 
I can provide a better advice if you give more details :)

Comment (andre-f-paggi) [reply papaparse.js]: Of course,

I tried adding information to the querystring of the blob url.

Ex:
blob:https://example.org/9115d58c-bcda-ff47-86e5-083e9a215304?papaworker=true

But the url goes 404 when I do that, it only accepts the url as is.

Comment (andre-f-paggi) [reply papaparse.js]: Well searching a little more, I can pass other information in the url, but the method is different, is via fragments: 

http://jsfiddle.net/thpf584n/1/

File: papaparse.js (hunk 4 above)
Comment (pokoli) [reply papaparse.js]: Forget about this comment.

File: papaparse.js (hunk 3 above)
Comment (pokoli) [reply papaparse.js]: Ok, I see the problem now. Lets forget about the suffix and use diretcly the Blob URL without SCRIPT_PATH_AS_BLOB_URL. If there are some problems with this design we should probably reintroduce the flag to enable blob loading in the future. 

Sorry for not giving the best advices.

**Comment by pokoli** [comment]
Could you please also remove the package-lock.json file?

**Comment by andre-f-paggi** [comment]
Hi, sorry for letting this PR open for so long, I completely forgot about it.

It is ok now?

File: papaparse.js (hunk 2 above)
Comment (pokoli) [inline papaparse.js]: Can we remove the SCRIPT_PATH_AS_BLOB_URL please?

Comment (andre-f-paggi) [reply papaparse.js]: Man, I don't know. If we remove this, it will always load as a blob url?

Honestly, I kind of lost the interest in making this PR, because it seems like I was the only one having this problem, it was very specific to my situation.

But if you still think it is still useful, I can try removing that.

Comment (pokoli) [reply papaparse.js]: Yes, we should load always as blob url. 

Do you think there is any drawback of doing so?

**Comment by pokoli** [comment]
Replaced by #599
